### PR TITLE
HSEARCH-3986 Test that Hibernate Search releases all file handles on close

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-3986

### CODE REVIEW

File: util/internal/integrationtest/backend/lucene/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/lucene/directory/TrackingDirectoryProvider.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package org.hibernate.search.util.impl.integrationtest.backend.lucene.directory;
+
+import org.hibernate.search.backend.lucene.lowlevel.directory.spi.DirectoryCreationContext;
+import org.hibernate.search.backend.lucene.lowlevel.directory.spi.DirectoryHolder;
+import org.hibernate.search.backend.lucene.lowlevel.directory.spi.DirectoryProvider;
+
+import org.apache.lucene.store.SingleInstanceLockFactory;
+
+
+public class TrackingDirectoryProvider implements DirectoryProvider {
```
Comment (fax4ever) [inline util/internal/integrationtest/backend/lucene/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/lucene/directory/TrackingDirectoryProvider.java:16]: Actually the scope of this class is not clear to me. I was thinking if this class is supposed to be used by some user should be documented somehow. If not, maybe should be moved to the tests.

File: integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/lifecycle/LuceneCleanupIT.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+						(BeanReference<TrackingDirectoryProvider>) beanResolver -> {
+							BeanHolder<DirectoryProvider> delegateHolder = beanResolver.resolve( DirectoryProvider.class,
+									"local-filesystem" );
+							return BeanHolder.of( new TrackingDirectoryProvider( delegateHolder.get(), tracker ) )
+									.withDependencyAutoClosing( delegateHolder );
+						} )
+				.setup();
+	}
+
+
```
Comment (fax4ever) [inline integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/lifecycle/LuceneCleanupIT.java:142]: 2 spaces

**Review by fax4ever — COMMENTED** [review]
just a question

File: util/internal/integrationtest/backend/lucene/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/lucene/directory/TrackingDirectoryProvider.java (hunk 1 above)
Comment (yrodiere) [reply util/internal/integrationtest/backend/lucene/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/lucene/directory/TrackingDirectoryProvider.java:20]: It is in test code. More precisely, utils for integration tests:

```
package org.hibernate.search.util.impl.integrationtest.backend.lucene.directory
```

I'll add a short comment.

**Comment by yrodiere** [comment]
Addressed the comments and merged. Thanks.

**Pull request merged**